packages/sanity/src/core/error/sentry/sentryErrorReporter.ts

```diff
@@ -172,9 +174,26 @@ export function getSentryErrorReporter(): ErrorReporter {
     return eventId ? {eventId} : null
   }
 
+  function isBufferedTransport(transport: Transport | undefined): transport is BufferedTransport {
+    return !!transport && 'setConsent' in transport && typeof transport.setConsent === 'function'
+  }
+
+  function enable() {
+    if (isBufferedTransport(client?.getTransport())) {
+      ;(client?.getTransport() as BufferedTransport).setConsent(true)
+    }
```

Comment:
Typescript can't know that this is idempotent, but assigning to a variable should do the trick, I believe
```suggestion
    const transport = client?.getTransport()
    if (isBufferedTransport(transport)) {
      transport.setConsent(true)
    }
```